src/api.v2/helpers/pipeline/pipelineConstruct/index.js

```diff
@@ -87,11 +88,23 @@ const withRecomputeDoubletScores = (processingConfig) => {
   return newProcessingConfig;
 };
 
+const getMetadataS3Path = async (experimentId) => {
+  const cellLevelMetadata = await new CellLevelMeta().getMetadataByExperimentIds([experimentId]);
+  if (cellLevelMetadata.length === 0) {
+    return null;
+  }
+  else if (cellLevelMetadata.length === 1) {
+    return cellLevelMetadata[0].id;
+  }
```

Comment:
You could also replace these checks with `return cellLevelMetadata[0]?.id || null`.

Reply:
And could do an explicit check first on the error that you throw of multiple cell level metadata files.

`if (cellLevelMetadata.length > 1) {
    throw ....
}`